core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -37,31 +33,19 @@
 public abstract class RetryingClient<I extends Request, O extends Response>
         extends SimpleDecoratingClient<I, O> {
 
-    private static final Logger logger = LoggerFactory.getLogger(RetryingClient.class);
-
     private final Supplier<? extends Backoff> backoffSupplier;
     private final RetryRequestStrategy<I, O> retryStrategy;
     private final int defaultMaxAttempts;
 
     /**
      * Creates a new instance that decorates the specified {@link Client}.
      */
-    protected RetryingClient(Client<? super I, ? extends O> delegate,
-                             RetryRequestStrategy<I, O> retryStrategy,
-                             Supplier<? extends Backoff> backoffSupplier) {
-        this(delegate, retryStrategy, backoffSupplier, Flags.defaultBackoffMaxAttempts());
-    }
-
-    /**
-     * Creates a new instance that decorates the specified {@link Client}.
-     */
-    protected RetryingClient(Client<? super I, ? extends O> delegate,
-                             RetryRequestStrategy<I, O> retryStrategy,
-                             Supplier<? extends Backoff> backoffSupplier,
-                             int defaultMaxAttempts) {
+    public RetryingClient(Client<? super I, ? extends O> delegate,
```

Comment:
No point of having a public constructor for an abstract class. How about `protected`?